### PR TITLE
Fix clear bug in edit sub component

### PR DESCRIPTION
**Pull request by hetec**

### CODE REVIEW

File: socobo-recipe-overview.html [hunk 1]
```diff
@@ -113,90 +120,210 @@
...
-      }
-      this.$.editDialog.recipe = recipe;
+    /**
+     *
+     * @param e
+     */
+    edit : function(e) {
+      //this.$.editDialog.recipe_inputs = [];
+      //this.$.editDialog.ingredient_inputs = [];
+      var eClone = (JSON.parse(JSON.stringify(e.detail)))
```
Comment (flashback2k14) [inline socobo-recipe-overview.html]: not working in socobo project.
error: JSON.stringify(e.detail) - Converting circular structure to JSON

File: ImageService.js [hunk 2]
```diff
@@ -1,19 +1,25 @@
...
+      id.src = data;
+    }
+  };
+
   this.encode = function (imageId) {
     var file = imageChooser.files[0];
     if(file === undefined || file === null){
       data = "";
-      build(imageId,"http://localhost:8080/components/socobo-recipe/placeholder.png");
+      build(imageId,"../placeholder.png");
```
Comment (flashback2k14) [inline ImageService.js]: not found in socobo project

**Comment by hetec** [comment]
Info:

Placeholder image was not found because the root of the path is not the current directory in gulp. Gulp begins its search in the server root, so the working path resolves to bower_components/socobo-recipe/placeholder.png

The problem with cloning the event object was not caused by the Json module. The cause was a cyclic reference of the ref attribute. Therefore it was not sufficient to create an own deep clone method because this would also create a stack overflow. The solution was to convert the ref to a string.
Currently the new self-implemented deep clone method is used. If some problem occurs it's always possible to switch back to the Json way.

**Comment by hetec** [comment]
Yes. Nice idea, so the ref ist more consistent

**Comment by flashback2k14** [comment]
i.O.

**Pull request merged**